app/src/main/java/helper/CourseRequestHandler.java

```diff
@@ -149,10 +157,52 @@ public void onFailure(Call<List<CourseSection>> call, Throwable t) {
     }
 
 
+    private List<CourseSection> resolve(Object response) {
+        List<CourseSection> courseSections = (List<CourseSection>) response;
+        List<Content> contents = new ArrayList<>();
+        for(CourseSection courseSection : courseSections) {
+            for (Module module : courseSection.getModules()) {
+                if(module.getContents()!=null) {
```

Comment:
Please store `module.getContents()` in a variable and check if that is `null` or not, pass the same variable to `addAll` method below.